ptsa/data/TimeSeriesX.py

```diff
@@ -111,43 +121,147 @@ def to_hdf(self, filename, mode='w'):
             if self.name is not None:
-                root.attrs['name'] = self.name.encode()
+                root.attrs['name'] = self.name.encode(encoding)
             if self.attrs is not None:
-                root.attrs['attrs'] = json.dumps(self.attrs).encode()
+                root.attrs['attrs'] = json.dumps(self.attrs).encode(encoding)
+
+    @classmethod
+    def from_hdf(cls, filename, decode_string_arrays=True, encoding='utf-8'):
+        """
+        Deserialize from HDF5 using :mod:`h5py`.
+        Parameters
+        ----------
+        filename : str
+        decode_string_arrays: bool
+            Arrays of bytes should be decoded into strings
+        encoding: str
+            Encoding scheme to use for decoding
+        Returns
+        -------
+        Deserialized instance
+        """
+        if h5py is None:  # pragma: nocover
+            raise RuntimeError("You must install h5py to load from HDF5")
+
+        with h5py.File(filename, 'r') as hfile:
+            dims = hfile['dims'][:]
+
+            root = hfile['/']
+
+            coords_group = hfile['coords']
+            names = json.loads(coords_group.attrs['names'].decode(encoding))
+            coords = dict()
+            for name in names:
+                data = coords_group[name]
+                data_has_fields = len(data.dtype) > 0
+                if ~data_has_fields and data.dtype.char == 'S':
+                    data = [s.decode(encoding) for s in np.atleast_1d(data)]
+                elif data_has_fields:
+                    # Determine what the final dtypes will be
+                    final_dtypes = []
+                    bytes_fields = []
+                    for i, field in enumerate(data.dtype.names):
+                        if data[field].dtype.kind != 'S':
+                            final_dtypes.append((field,
+                                                 data[field].dtype.str))
+                        else:
+                            final_dtypes.append((field, 'U256'))
+                            bytes_fields.append(field)
+                            
+                    # Update dtypes of the data. This will coerce the
+                    # bytes fields to unicode automatically
+                    data = np.array(data).astype(final_dtypes).view(np.recarray)
+                coords[name] = data
+            name = root.attrs.get('name', None)
+            if name is not None:
+                name = name.decode(encoding)
+            attrs = root.attrs.get('attrs', None)
+            if attrs is not None:
+                attrs = json.loads(attrs.decode(encoding))
+
+            array = cls.create(hfile['data'].value, None, coords=coords,
+                               dims=[dim.decode() for dim in dims],
+                               name=name, attrs=attrs)
+            return array
 
     @classmethod
-    def from_hdf(cls, filename):
-        """Load from an HDF5 file.
+    def from_hdf_legacy(cls, filename):
```

Comment:
I don't like naming things like this. I'd rather there be a way in the `from_hdf` method to detect that it's the older format and load it that way (complete with deprecation warning). Then you can make this a private method (`_from_hdf_legacy` or whatever) to not make the public method too long (if desired).